cmd/oras/upload.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/oci"
+	"oras.land/oras-go/v2/registry"
+	"oras.land/oras/cmd/oras/internal/display"
+	"oras.land/oras/cmd/oras/internal/option"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+)
+
+const (
+	directory = "output"
+)
+
+type uploadOptions struct {
+	option.Common
+	To option.Remote
+}
+
+func uploadCmd() *cobra.Command {
+	var opts uploadOptions
+	cmd := &cobra.Command{
+		Use:     "upload [flags] <source>{:<tag>|@<digest>} [<source>{:<tag>|@<digest>}][...]]",
+		Aliases: []string{"upload"},
+		Short:   "[Preview] upload multiple artifacts",
+		Long: `[Preview] upload multiple artifacts
+
+** This command is in preview and under development. **
+
+Example - uplad artifacts
+  oras upload localhost:5000/owner/this:v1 localhost:6000/owner/that:v1
+`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return option.Parse(&opts)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runupload(opts, args[0])
+		},
+	}
+	option.ApplyFlags(&opts, cmd.Flags())
+	return cmd
+}
+
+func runupload(opts uploadOptions, destination string) error {
+	ctx, _ := opts.SetLoggerLevel()
+
+	var uniqueImages []registry.Reference
```

Comment:
this is an array of repositories?
```suggestion
	var repos []registry. Reference
```